ruma-federation-api/src/keys/claim_keys/v1.rs

```diff
@@ -0,0 +1,38 @@
+//! [POST /_matrix/federation/v1/user/keys/claim](https://matrix.org/docs/spec/server_server/r0.1.4#id91)
+
+use std::collections::BTreeMap;
+
+use ruma_api::ruma_api;
+use ruma_identifiers::{DeviceId, DeviceKeyAlgorithm, DeviceKeyId, UserId};
+use serde::{Deserialize, Serialize};
+
+ruma_api! {
+    metadata: {
+        description: "Claims one-time keys for use in pre-key messages.",
+        method: POST,
+        name: "claim_keys",
+        path: "/_matrix/federation/v1/user/keys/claim",
+        rate_limited: false,
+        requires_authentication: false,
+    }
+
+    request: {
+        /// The keys to be claimed.
+        /// #[ruma_api(path)]
+        one_time_keys: BTreeMap<UserId, BTreeMap<Box<DeviceId>, DeviceKeyAlgorithm>>,
+    }
+
+    response: {
+        /// One-time keys for the queried devices
+        one_time_keys: BTreeMap<UserId, BTreeMap<Box<DeviceId>, BTreeMap<DeviceKeyId, KeyObject>>>
+    }
+}
+
+/// A key and its signature
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct KeyObject {
```

Comment:
Please make non-exhaustive and provide a constructor.

```suggestion
#[non_exhaustive]
pub struct KeyObject {
```